README.md

```diff
@@ -340,7 +340,32 @@ We also handle type promotion by explicitly coding for it. This is done in the a
 ### Testing
 
 Running tests will generate a `test-report.html` in the root dir.
-**TODO** Explain test organizatian and expression tables
 
-Evaluating within a test should be done by calling `generalEvaluate`.
-This will test both sync and async implementation if possible. 
+The testing environment is set up to do a lot of tests with little code.
+The files responsible for fluent behaviour reside in `test/util`.  
+Most test can be run by running the `runTestTable` method in `test/util/utils.ts`.
+This method expects a TruthTable. Multiple test are run over a TruthTable (one for every line).
+A truth table may contain aliases if the aliases are also provided
+(Some handy aliases reside in `test/util/Aliases.ts`).
+This means that when testing something like `"3"^^xsd:integer equals "3"^^xsd:integer` is `"true"^^xsd:boolean`.
+We would write a small table and test it like this:
+```ts
+import { bool, merge, numeric } from './util/Aliases';
+import { Notation } from './util/TruthTable';
+import { runTestTable } from './util/utils';
+runTestTable({
+   arity: 2,
+   operation: '=',
+   aliases: merge(numeric, bool),
+   notation: Notation.Infix,
+   testTable: `
+    3i 3i = true
```

Comment:
Nitpick. Let's put the testTable argument first, and add some more examples to highlight the use of it. We want to convince future developers of this approach!

Reply:
What expamples do you mean? The same test but new lines to show how multiple tests in the table are created?

Reply:
Yes exactly! It's easier to abstract from multiple demonstrations :)